.github/workflows/robot/internal/bot/backport.go

```diff
@@ -0,0 +1,263 @@
+/*
+Copyright 2022 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package bot
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"log"
+	"net/url"
+	"os/exec"
+	"path"
+	"regexp"
+	"sort"
+	"strconv"
+	"strings"
+	"text/template"
+
+	"github.com/gravitational/teleport/.github/workflows/robot/internal/github"
+
+	"github.com/gravitational/trace"
+)
+
+// Backport will create backport Pull Requests (if requested) when a Pull
+// Request is merged.
+func (b *Bot) Backport(ctx context.Context) error {
+	if !b.c.Review.IsInternal(b.c.Environment.Author) {
+		return trace.BadParameter("automatic backports are only supported for internal contributors")
+	}
+
+	pull, err := b.c.GitHub.GetPullRequest(ctx,
+		b.c.Environment.Organization,
+		b.c.Environment.Repository,
+		b.c.Environment.Number)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	// Extract backport branches names from labels attached to the Pull
+	// Request. If no backports were requested, return right away.
+	branches := findBranches(pull.UnsafeLabels)
+	if len(branches) == 0 {
+		return nil
+	}
+
+	var rows []row
+	var failed bool
+
+	// Loop over all requested backport branches and create backport branch and
+	// GitHub Pull Request.
+	for _, base := range branches {
+		head := fmt.Sprintf("bot/backport-%v-%v", b.c.Environment.Number, base)
+
+		r := row{
+			Branch: base,
+		}
+
+		// Create and push git branch for backport to GitHub.
+		err := b.createBackportBranch(ctx,
+			b.c.Environment.Organization,
+			b.c.Environment.Repository,
+			b.c.Environment.Number,
+			base,
+			pull,
+			head,
+		)
+		if err != nil {
+			failed = true
+			rows = append(rows, r)
+			continue
+		}
```

Comment:
Looks like we're dropping the error here (and in CreatePullRequest?). What if an `error` field was added to the `row` struct and included in the template?

Reply:
I had that in one of my earlier iterations, you can see it here.

https://github.com/russjones/teleport/pull/47

I didn't find it that useful because the commit isn't that interesting, it's the conflict that is, and that's harder to extract.

That being said, if you think the commit would be useful, I can add it back or at least maybe log the error?

Reply:
Yeah, that error isn't really useful. I was thinking something strange could happen with one of the multiple git calls in `createBackportBranch` (fetch, checkout, rebase, push) and we wouldn't know which step failed. We're logging the config call but the others are dropped.

Reply:
Ah, good catch. Fixed I log the error and link to the backport run logs. You can see an example here: https://github.com/russjones/teleport/pull/90.